### DISCUSSION TITLE
Display black text

### DISCUSSION THREAD

**Discussion by commputethis** [pixelit-project/PixelIt#332] [Q&A]
Is it possible to reverse the text to have the lights off for text and the rest of the display on?  Basically creating black text.  The use case is that I have the remaining print time from my 3D printer displayed and have the color change based on the color of the filament being printed with.  It works great for everything I have used except black, of course.  I could change the color of the text, but think it would be nice to basically have the text outlined vs using a different color. Hope that makes sense.

Below is a link to a short clip of my PixelIt.
https://photos.app.goo.gl/61sdvYE5a7D2AVsU6

**Comment by rliegmann** [comment]
Do you mean that the text is inverted, so to speak the negative of the text?

**Reply by commputethis** [reply]
I think what you are asking is what I'm thinking. Basically, the text to be black (lights off) and the rest of the LEDs on.

**Reply by rliegmann** [reply]
Like this: ??
![Bildschirmfoto 2024-02-09 um 23 11 04](https://github.com/pixelit-project/PixelIt/assets/16445478/acb9f3d9-5604-4eb0-9ab3-198ae631e6a6)

**Comment by commputethis** [comment]
Yes

**Comment by rliegmann** [comment]
According to research, it's not that easy.
PixelIt uses the Adafruit GFX library with its own font. As soon as a custom font is used, it is no longer possible to set the background color.
For text that doesn't scroll, you could first switch all pixels to one color and then display the text. As soon as a text needs to be scrolled, the solution no longer works.
Alternativ kann man ein Bitmap senden mit dem Text.

I hope it answers your question. Maybe others have a solution. :-)
